scripts/validate-requirements.sh

```diff
@@ -0,0 +1,221 @@
+    fi
+}
+# Validate the Git installation and version
+function validate_git(){
+    if [ ! "$(command -v git)" ]; then
+        echo "git not found."
+        echo "Visit https://git-scm.com/book/en/v2/Getting-Started-Installing-Git and follow the instructions to install Git."
+        ERRORS+=$'git not found.\n'
+    else
+        GIT_CURRENT_VERSION=$(git version | awk '{print $3}')
+        compare_version "$GIT_CURRENT_VERSION" "$GIT_VERSION"
+        if [ $? -eq 2 ]; then
+            echo "An error was found with the git installation."
+            echo "Version required is $GIT_VERSION"
+            echo "Visit https://git-scm.com/book/en/v2/Getting-Started-Installing-Git and follow the instructions to install Git."
+            ERRORS+=$'git version is incompatible.\n'
+        fi
+    fi
+
+    if ! git config init.defaultBranch | grep "main" >/dev/null ; then
+        echo "git defaul branch must be configured."
+        echo "See the instructions at https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/docs/TROUBLESHOOTING.md#default-branch-setting ."
+        ERRORS+=$'git default branch must be configured.\n'
+    fi
+}
+
+# Validate the Configuration of the Gcloud CLI
+function validate_gcloud_configuration(){
+    gcloud config get-value account 2> end-user-credential-output.txt >/dev/null
+    if grep -q 'unset' end-user-credential-output.txt ; then
+        echo "You must configure an End User Credential."
+        echo "Visit https://cloud.google.com/sdk/gcloud/reference/auth/login and follow the instructions to authorize gcloud to access the Cloud Platform with Google user credentials."
+        ERRORS+=$'gcloud end user credential not configured.\n'
+    fi
+    rm -Rf end-user-credential-output.txt
+
+    gcloud auth application-default print-access-token 2> application-default-credential-output.txt >/dev/null
+    if grep -q 'Could not automatically determine credentials' application-default-credential-output.txt ; then
+        echo "You must configure an Application Default Credential."
+        echo "Visit https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login and follow the instructions to authorize gcloud to access the Cloud Platform with Google user credentials."
+        ERRORS+=$'gcloud application default credential not configured.\n'
+    fi
+    rm -Rf application-default-credential-output.txt
+}
+
+# Function to validate the Configuration of the Gcloud CLI
+function validate_credential_roles(){
+    check_org_level_roles "$END_USER_CREDENTIAL" "$ORGANIZATION_ID"
+    check_billing_account_roles "$END_USER_CREDENTIAL" "$BILLING_ACCOUNT"
+}
+
+# Verifies wheter a user is assigned to the expected Orgzanization Level roles
+function check_org_level_roles(){
+
+    gcloud organizations get-iam-policy "$2" \
+    --filter="bindings.members:$1" \
+    --flatten="bindings[].members" \
+    --format="table(bindings.role)" 2>/dev/null > org-level-roles-output.txt
```

Comment:
is it possible to save this data in memory?

Reply:
Yes, just found a way for doing so. Now I'll replicate to the other cases.